sim/paladin/guardian_of_ancient_kings.go

```diff
@@ -0,0 +1,226 @@
+			if oldStacks > 0 {
+				paladin.DisableDynamicStatDep(sim, strDepByStackCount[oldStacks])
+			}
+
+			if newStacks > 0 {
+				paladin.EnableDynamicStatDep(sim, strDepByStackCount[newStacks])
+			}
+		},
+	})
+
+	ancientFuryMinDamage, ancientFuryMaxDamage :=
+		core.CalcScalingSpellEffectVarianceMinMax(proto.Class_ClassPaladin, 0.2366, 0.3)
+	numTargets := paladin.Env.GetNumTargets()
+	results := make([]*core.SpellResult, numTargets)
+
+	ancientFury := paladin.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 86704},
+		SpellSchool:    core.SpellSchoolHoly,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          core.SpellFlagAPL,
+		ClassSpellMask: SpellMaskAncientFury,
+
+		MaxRange: 10,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				NonEmpty: true,
+			},
+		},
+
+		DamageMultiplier: 1,
+		CritMultiplier:   paladin.DefaultSpellCritMultiplier(),
+		ThreatMultiplier: 1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := sim.Roll(ancientFuryMinDamage, ancientFuryMaxDamage) +
+				0.061*spell.SpellPower()
+
+			// Deals X Holy damage per application of Ancient Power,
+			// divided evenly among all targets within 10 yards.
+			baseDamage *= float64(ancientPower.GetStacks())
+			baseDamage /= float64(numTargets)
+
+			for idx := int32(0); idx < numTargets; idx++ {
+				currentTarget := sim.Environment.GetTargetUnit(idx)
+				results[idx] = spell.CalcDamage(sim, currentTarget, baseDamage, spell.OutcomeMagicHitAndCrit)
+			}
+
+			for idx := int32(0); idx < numTargets; idx++ {
+				spell.DealDamage(sim, results[idx])
+			}
+		},
+	})
+
+	goakAura := paladin.RegisterAura(core.Aura{
+		Label:    "Guardian of Ancient Kings",
+		ActionID: core.ActionID{SpellID: 86698},
+		Duration: duration,
+
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if !result.Landed() || spell.ClassSpellMask&SpellMaskCanTriggerAncientPower == 0 {
+				return
+			}
+
+			ancientPower.AddStack(sim)
+		},
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			ancientPower.Activate(sim)
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			paladin.AncientGuardian.Pet.Disable(sim)
+			ancientFury.Cast(sim, paladin.CurrentTarget)
+			ancientPower.Deactivate(sim)
+		},
+	})
+
+	return paladin.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 86150},
+		Flags:          core.SpellFlagAPL,
+		ProcMask:       core.ProcMaskEmpty,
+		ClassSpellMask: SpellMaskGuardianOfAncientKings,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				NonEmpty: true,
+			},
+			CD: core.Cooldown{
+				Timer:    paladin.NewTimer(),
+				Duration: time.Minute * 5,
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, unit *core.Unit, spell *core.Spell) {
+			goakAura.Activate(sim)
+			paladin.AncientGuardian.EnableWithTimeout(sim, paladin.AncientGuardian, duration)
```

Comment:
You prob just need to enable the pet and dont need the timeout as you're manually handling the deactivation with the goak aura